scripts/run_backend_tests.py

```diff
@@ -242,6 +242,10 @@ def main(args=None):
     """Run the tests."""
     parsed_args = _PARSER.parse_args(args=args)
 
+    # Make sure that third-party libraries are up-to-date before running tests,
+    # otherwise import errors may result.
+    install_third_party_libs.main()
```

Comment:
Considering `install_third_party_libs` itself runs `setup.main` and `setup_gae.main` I think we don't need to re-run them overe here.

(Also, there are a lot of unwanted things we will be checking like lint parser, installing node modules, installing git hooks, etc. Maybe in the future we can move `pip_dependencies` to install_third_party  module and run that script?)

Reply:
Hi @DubeySandeep, thanks for catching that! I have removed those two lines and the correpsonding imports. 

I'm afraid I don't understand your second paragraph -- could you please clarify? Where are we checking that, and what exactly is your concern? If you are worried about the additional overhead of running install_third_party_libs in this file, I can say that it takes < 10 seconds on my computer.

Reply:
I thought we wanted to install all third-party libs (mostly python) over here, so we are running `install_third_party_libs`. But we have `install_third_party` module which use to download all the required libs except the pip modules so maybe we can move pip module installation steps to `install_third_party` and use `install_third_party.main()` over here that way we can avoid all other unwanted things like installing node modules, githooks, configuring pylint, etc. which aren't required for running backend tests


> If you are worried about the additional overhead of running install_third_party_libs in this file, I can say that it takes < 10 seconds on my computer.

Thanks for the info @seanlip, I think it's fine but it can be a bit bigger if we update node modules or js libs (which are not required for backend test) but it won't be that big.

Reply:
What I did on the e2e side was to add `--skip-install` flag (https://github.com/oppia/oppia/blob/develop/scripts/run_e2e_tests.py#L257), the benefit was to reduce the amount of noise that the install script makes on CircleCI.

Reply:
I'm actually starting to get a bit confused with all the suggestions here, sorry (more specifically, trying to figure out what exact follow-up actions should be taken).

@DubeySandeep @kevinlee12 would you mind filing issues with any remaining changes that you would like made to the scripts? We can continue the discussion in those threads. Thanks!